### PR TITLE
User list UI

### PR DESCRIPTION
**Pull request by harshup20**
issue fixed of the database

https://user-images.githubusercontent.com/68155750/124113986-ae9f9200-da89-11eb-90fa-4db671ba73c9.mp4

### CODE REVIEW

File: lib/datasource/remote/implementation/user_datasource_imp.dart [hunk 1]
```diff
@@ -28,4 +34,49 @@ class UserDataSourceImpl extends UserDataSource {
...
     }
   }
+
+  @override
+  Future<List<UserDomain>> getUsersFromHive() async {
+
+    // return users hive box
+    try{
+ final response = await _hiveRequest.getUsers();
+ UserResponse userResponse = UserResponse(response);
```
Comment (mayyuu) [inline lib/datasource/remote/implementation/user_datasource_imp.dart:44]: UserResponse class is not handling response returned from hive,there is not need to use this class here

File: lib/datasource/remote/implementation/user_datasource_imp.dart [hunk 2]
```diff
@@ -28,4 +34,49 @@ class UserDataSourceImpl extends UserDataSource {
...
+    }
+
+  }
+
+  @override
+  Future insertUsers(List<UserDomain> users) async {
+    try {
+      // return users hive box
+      final response = await _hiveRequest.addUser(users);
+      UserResponse userResponse = UserResponse(response);
```
Comment (mayyuu) [inline lib/datasource/remote/implementation/user_datasource_imp.dart:65]: UserResponse class is not handling response returned from hive,there is not need to use this class here

File: lib/presentation/user_list/all_user_list/base_list_view.dart [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+        model.userListScrollController.addListener(() async {
+          if (model.userListScrollController.position.maxScrollExtent ==
+                  model.userListScrollController.position.pixels &&
+              !model.busy) {
+            model.setBusy(true);
+            print('List End: Loading more user');
+            await model
+                .getUserList(params: GetUsersUseCaseParams(model.page))
+                .then((response) {
+              model.userList.addAll(response);
```
Comment (mayyuu) [inline lib/presentation/user_list/all_user_list/base_list_view.dart:31]: Viewmodel class should contain business logic which you have added inside presentation layer please update this and handle inside your viewmodel -> then((response) {
              model.userList.addAll(response);

File: lib/presentation/user_list/all_user_list/list_view_model.dart [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  set userList(List<UserItem> value) {
+    _userList = value;
+    notifyListeners();
+  }
+
+
+  void refresh() {
+    notifyListeners();
+  }
+  void selectCard(UserItem userItem,{BuildContext context,
```
Comment (mayyuu) [inline lib/presentation/user_list/all_user_list/list_view_model.dart:39]: BuildContext comes under presentation layer which we cannot use it inside viewmodel class as this handels only our business logic ,please update

File: lib/presentation/user_list/all_user_list/list_view_model.dart [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    AddHiveUsersUseCaseParams getHiveUsersUseCaseParams })async{
+    addUsersUseCase = Provider.of(context, listen: false);
```
Comment (mayyuu) [inline lib/presentation/user_list/all_user_list/list_view_model.dart:41]: addUsersUseCase = Provider.of(context, listen: false); never to use class directly inside your functions ,please init this addUserUseCase inside your constructor as per DI(Dependency Injection).

File: lib/presentation/user_list/user_list_view_mobile.dart [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_paginator/flutter_paginator.dart';
+import 'package:git_users/presentation/user_list/selected_user/selected_user_view.dart';
+import 'package:git_users/presentation/user_list/user_list_view_model.dart';
+import 'package:git_users/presentation/base/view/git_user_landing_base_view.dart';
+import 'package:git_users/presentation/user_list/all_user_list/base_list_view.dart';
+
+class UserListViewMobile extends UserListBaseModelWidget<UserListViewModel>{
+  UserListViewMobile();
```
Comment (mayyuu) [inline lib/presentation/user_list/user_list_view_mobile.dart:11]: AS we are calling UserListViewMobile inside our view class ,so there is no need to call this constructor here..we can remove it from here

File: lib/data/repository/hive_repository_impl.dart [hunk 7]
```diff
@@ -0,0 +1,19 @@
+// import 'package:git_users/data/datasources/local/user_local_datasource.dart';
+// import 'package:git_users/domain/model/user_domain.dart';
+// import 'package:git_users/domain/repository/hive_repository.dart';
+//
+// class HiveRepositoryImpl extends HiveRepository {
```
Comment (naikdp7) [inline lib/data/repository/hive_repository_impl.dart:5]: please remove this file

File: lib/datasource/remote/providers/rest/request/local/hive_request.dart [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      Hive.init(appDocumentDir.path);
+      Hive.registerAdapter(UserAdapter());
+      await Hive.openBox(Strings.userBox);
+      return true;
+    } on Exception catch (e) {
+      print(e);
+      return false;
+    }
+  }
+  Future addUser(List<UserDomain> users) async{
```
Comment (naikdp7) [inline lib/datasource/remote/providers/rest/request/local/hive_request.dart:24]: if i deselect one user from selected tab then also you are calling addUser?we need to have remove user method right?

File: lib/presentation/user_list/all_user_list/list_view_model.dart [hunk 9]
```diff
@@ -49,28 +45,44 @@ class BaseListViewModel extends BaseViewModel {
...
+        setBusy(true);
+        print('List End: Loading more user');
+        await getUserList(params: GetUsersUseCaseParams(page))
+            .then((response) {
+          userList.addAll(response);
+        });
+        setBusy(false);
+      }
+    });
+  }
```
Comment (mayyuu) [inline lib/presentation/user_list/all_user_list/list_view_model.dart:87]: Previous code was correct where listener was added inside view class -> userListScrollController.addListener(()..you only need to add your business logic code i.e ->

setBusy(true);
        print('List End: Loading more user');
        await getUserList(params: GetUsersUseCaseParams(page))
            .then((response) {
          userList.addAll(response);
        });
        setBusy(false);
        
       inside your viewmodel class.please update this

File: lib/presentation/user_list/all_user_list/list_view_model.dart [hunk 10]
```diff
@@ -64,25 +85,20 @@ class BaseListViewModel extends BaseViewModel {
...
     setBusy(false);
-    // return result;
   }
-  void onScroll(){
-    userListScrollController.addListener(() async {
-      if (userListScrollController.position.maxScrollExtent ==
-          userListScrollController.position.pixels &&
-          !busy) {
+  void onScroll()async{
         setBusy(true);
```
Comment (mayyuu) [inline lib/presentation/user_list/all_user_list/list_view_model.dart:91]: We are handling setBusy inside -> getUserList function ,so there is no need to add and handel it here

File: lib/presentation/user_list/all_user_list/list_view_model.dart [hunk 11]
```diff
@@ -64,25 +85,20 @@ class BaseListViewModel extends BaseViewModel {
...
         print('List End: Loading more user');
         await getUserList(params: GetUsersUseCaseParams(page))
             .then((response) {
           userList.addAll(response);
         });
         setBusy(false);
```
Comment (mayyuu) [inline lib/presentation/user_list/all_user_list/list_view_model.dart:97]: same setBusy(false) we are handling inside getUserList function so we can remove it from here

**Review by naikdp7 — APPROVED** [review]
Approved

**Pull request merged**